setup.py

```diff
@@ -147,7 +147,7 @@ def run(self):
             # Required to log artifacts and models to AWS S3 artifact locations
             "boto3",
             # Required to log artifacts and models to GCS artifact locations
-            "google-cloud-storage>=1.30.0",
+            "google-cloud-storage>=2.12.0",
```

Comment:
Is < 2.12 missing required features?

Reply:
- `XMLMPUContainer` class from `google.resumable_media.requests` [dependency](https://github.com/googleapis/google-resumable-media-python/blob/v2.6.0/google/resumable_media/requests/upload.py#L560). Without this, we need to build the xml request body ourselves (e.g. [this](https://cloud.google.com/storage/docs/xml-api/post-object-complete)).
- [some headers added](https://github.com/googleapis/python-storage/blob/v2.12.0/google/cloud/storage/_helpers.py#L618-L621) on mpu create request

Reply:
mpu support is quite new on the python gcs client

Reply:
Got it.

> Without this, we need to build the xml request body ourselves

This sounds painful.

Reply:
Is it possible to enable mpu for GCS only when `XMLMPUContainer` is importable?

Reply:
that's a good idea, let me think how to make it clean